### PR TITLE
[redefine storage setup] Redefining setup cloud storage link (store/storage)

### PR DESCRIPTION
**Pull request by lucasncabral**
The ml-git code has been modified to work with `storage` instead of `store`.  Deprecated `store` commands have been removed.  The user must use the `storage` command instead of old deprecated `store`.

```
Usage: ml-git repository [OPTIONS] COMMAND [ARGS]...

  Management of this ml-git repository.

Options:
  --help  Show this message and exit.

Commands:
  config   Configuration of this ml-git repository
  gc       Cleanup unnecessary files and optimize the use of the disk space.
  init     Initialiation of this ml-git repository
  remote   Configure remote ml-git metadata repositories.
  storage  Storage management for this ml-git repository.
  update   This command will update all ml-entities metadata repository.
```

Logic has also been added to allow automatic updating of an old project. This logic changes the keys that link to the storage in the entitiy config and spec files.

Config and spec:

```
dataset:
  git: ''
model:
  git: ''
storage: <- Changed from store to storage here
  s3:
    mlgit-datasets:
      aws-credentials:
        profile: default
      region: us-east-1
```

```
dataset:
  categories:
  - test
  manifest:
    files: MANIFEST.yaml
    storage: s3h://minio1 <- Changed from store to storage here
  mutability: strict
  name: dataset-ex
  version: 1
```

### CODE REVIEW

File: ml_git/file_system/local.py [hunk 1]
```diff
@@ -907,59 +907,59 @@ def export_tag(self, metadata_path, tag, bucket, retry):
             return
 
         manifest = spec[self.__repo_type]['manifest']
-        store = store_factory(self.__config, manifest['store'])
-        if store is None:
-            log.error('No store for [%s]' % (manifest['store']), class_name=LOCAL_REPOSITORY_CLASS_NAME)
+        storage = storage_factory(self.__config, manifest[STORAGE_KEY])
+        if storage is None:
+            log.error('No storage for [%s]' % (manifest[STORAGE_KEY]), class_name=LOCAL_REPOSITORY_CLASS_NAME)
```
Comment (GustavoMaciel) [inline ml_git/file_system/local.py:912]: The same message was used below, maybe put it into the messages dictionary to be reused.
You can also look for other log usages in this PR, there's a bunch of hardcoded messages that could go to the dictionary.

File: ml_git/ml_git_message.py [hunk 2]
```diff
@@ -11,13 +11,25 @@
...
+      storage: s3h://some-bucket
     name: datasets-ex
     version: 5
 '''
 
 output_messages = {
     'DEBUG_REMOVE_REMOTE': 'Removing remote from local repository [%s]',
+    'DEBUG_BUILDING_STORAGE_LOG': 'Building the storage.log with these added files',
+    'DEBUG_OBJECT_ALREADY_IN_STORAGE': 'Object [%s] already in %s storage',
+    'DEBUG_STORE_AND_BUCKET': 'Store [%s] ; bucket [%s]',
```
Comment (GustavoMaciel) [inline ml_git/ml_git_message.py:23]: Shouldn't this have storage instead of store?

**Review by GustavoMaciel — COMMENTED** [review]
Check if the comments I made also make sense for you

File: ml_git/ml_git_message.py (hunk 2 above)
Comment (gilMars) [inline ml_git/ml_git_message.py:23]: Store or Storage?

File: ml_git/file_system/local.py (hunk 1 above)
Comment (lucasncabral) [reply ml_git/file_system/local.py]: This message and the others that were changed in this PR were moved to the centralized message file in 847ad30.

File: ml_git/ml_git_message.py (hunk 2 above)
Comment (lucasncabral) [reply ml_git/ml_git_message.py]: This message was updated on 847ad30.

Comment (lucasncabral) [reply ml_git/ml_git_message.py]: Check previous comment.

**Review by GustavoMaciel — APPROVED** [review]

File: ml_git/storages/multihash_storage.py [hunk 3]
```diff
@@ -7,11 +7,13 @@
...
 from ml_git import log
-from ml_git.constants import MULTI_HASH_STORE_NAME
+from ml_git.constants import MULTI_HASH_STORAGE_NAME
 from multihash import multihash
 
+from ml_git.ml_git_message import output_messages
 
-class MultihashStore(object):
+
+class MultihashStorage(object):
```
Comment (andersonpablosilva) [inline ml_git/storages/multihash_storage.py:16]: Update constant MULTI_HASH_STORAGE_NAME value.

File: ml_git/storages/s3_storage.py [hunk 4]
```diff
@@ -13,42 +13,42 @@
...
+from ml_git.constants import STORAGE_FACTORY_CLASS_NAME, S3STORAGE_NAME, S3_MULTI_HASH_STORAGE_NAME, StorageType
 from ml_git.ml_git_message import output_messages
-from ml_git.storages.multihash_store import MultihashStore
-from ml_git.storages.store import Store
+from ml_git.storages.multihash_storage import MultihashStorage
+from ml_git.storages.storage import Storage
 
 
-class S3Store(Store):
+class S3Storage(Storage):
```
Comment (andersonpablosilva) [inline ml_git/storages/s3_storage.py:22]: Update constant S3STORAGE_NAME value.

File: ml_git/utils.py [hunk 5]
```diff
@@ -19,7 +19,7 @@
 from ruamel.yaml.compat import StringIO
 
 from ml_git import log
-from ml_git.constants import SPEC_EXTENSION, CONFIG_FILE, EntityType, ROOT_FILE_NAME
+from ml_git.constants import SPEC_EXTENSION, CONFIG_FILE, EntityType, ROOT_FILE_NAME, OLD_STORAGE_KEY, STORAGE_KEY
```
Comment (andersonpablosilva) [inline ml_git/utils.py:22]: Maybe we can call it V1_STORAGE_KEY instead OLD_STORAGE_KEY. Same for OLD_MODELS_KEY and OLD_DATASETS_KEY.

File: ml_git/utils.py [hunk 6]
```diff
@@ -68,15 +68,15 @@ def json_load(file):
 
 
 def check_spec_file(file, hash):
-    modified = False
     if OLD_DATASETS_KEY in hash:
         hash[EntityType.DATASETS.value] = hash.pop(OLD_DATASETS_KEY)
-        modified = True
     if OLD_MODELS_KEY in hash:
```
Comment (andersonpablosilva) [inline ml_git/utils.py:73]: Can we move OLD_MODELS_KEY and OLD_DATASETS_KEY definitions to constants.py?

**Review by andersonpablosilva — CHANGES_REQUESTED** [review]
Please, take a look at the comments I've made.

File: ml_git/api.py [hunk 7]
```diff
@@ -224,25 +224,25 @@ def init(entity):
...
+            bucket_type (str, optional): Storage type (s3h, azureblobh or gdriveh) [default: s3h].
             credentials (str, optional): Name of the profile that stores the credentials or the path to the credentials.
             global_configuration (bool, optional): Use this option to set configuration at global level [default: False].
-            endpoint_url (str, optional): Store endpoint url.
+            endpoint_url (str, optional): Storage endpoint url.
     """
 
-    if bucket_type not in StoreType.to_list():
+    if bucket_type not in StorageType.to_list():
         log.error('Aqui2')
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/api.py]: This line was removed on 2f34d76.

File: ml_git/constants.py [hunk 8]
```diff
@@ -14,19 +14,19 @@
 HASH_FS_CLASS_NAME = 'HashFS'
 LOCAL_REPOSITORY_CLASS_NAME = 'Local Repository'
 MULTI_HASH_CLASS_NAME = 'Multihash'
-STORE_FACTORY_CLASS_NAME = 'Store Factory'
+STORAGE_FACTORY_CLASS_NAME = 'Store Factory'
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/constants.py]: The constant values ​​for the classes whose names have been changed were updated at 2f34d76.

File: ml_git/file_system/local.py [hunk 9]
```diff
@@ -907,59 +906,59 @@ def export_tag(self, metadata_path, tag, bucket, retry):
...
+        storage_dst_type = 's3://{}'.format(bucket_name)
+        storage_dst = storage_factory(self.__config, storage_dst_type)
+        if storage_dst is None:
+            log.error(output_messages['ERROR_WITHOUT_STORAGE'] % storage_dst_type, class_name=LOCAL_REPOSITORY_CLASS_NAME)
             return
         manifest_file = MANIFEST_FILE
         manifest_path = os.path.join(metadata_path, categories_path, manifest_file)
         files = yaml_load(manifest_path)
-        log.info('Exporting tag [{}] from [{}] to [{}].'.format(tag, manifest['store'], store_dst_type),
+        log.info('Exporting tag [{}] from [{}] to [{}].'.format(tag, manifest[STORAGE_KEY], storage_dst_type),
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/file_system/local.py]: Done 2f34d76.

File: ml_git/storages/azure_storage.py [hunk 10]
```diff
@@ -8,45 +8,45 @@
...
-from ml_git.storages.multihash_store import MultihashStore
-from ml_git.storages.store import Store
+from ml_git.constants import AZURE_STORAGE_NAME, StorageType, STORAGE_KEY
+from ml_git.ml_git_message import output_messages
+from ml_git.storages.multihash_storage import MultihashStorage
+from ml_git.storages.storage import Storage
 
 
-class AzureMultihashStore(Store, MultihashStore):
+class AzureMultihashStorage(Storage, MultihashStorage):
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/storages/azure_storage.py:17]: Done 2f34d76.

File: ml_git/storages/google_drive_storage.py [hunk 11]
```diff
@@ -16,19 +16,20 @@
...
-from ml_git.storages.store import Store
+from ml_git.constants import GDRIVE_STORAGE
+from ml_git.ml_git_message import output_messages
+from ml_git.storages.multihash_storage import MultihashStorage
+from ml_git.storages.storage import Storage
 from ml_git.utils import ensure_path_exists
 
 
-class GoogleDriveStore(Store):
+class GoogleDriveStorage(Storage):
```
[the remark this answers is not in the document]
Comment (lucasncabral) [reply ml_git/storages/google_drive_storage.py:26]: Done 2f34d76.

File: ml_git/storages/multihash_storage.py (hunk 3 above)
Comment (lucasncabral) [reply ml_git/storages/multihash_storage.py:16]: Done 2f34d76.

File: ml_git/storages/s3_storage.py (hunk 4 above)
Comment (lucasncabral) [reply ml_git/storages/s3_storage.py:22]: Done 2f34d76.

File: ml_git/utils.py (hunk 5 above)
Comment (lucasncabral) [reply ml_git/utils.py]: The variables referring to the keys in version 1 of ml-git have been renamed.

File: ml_git/utils.py (hunk 6 above)
Comment (lucasncabral) [reply ml_git/utils.py]: Moved in 18d35e7.

**Review by gilMars — APPROVED** [review]

**Review by andersonpablosilva — APPROVED** [review]

File: ml_git/utils.py [hunk 12]
```diff
@@ -320,24 +320,39 @@ def update_directories_to_plural(root_path, old_value, new_value):
...
-        update_now = input(output_messages['INFO_AKS_IF_WANT_UPDATE_PROJECT']).lower()
-        if update_now in ['yes', 'y']:
-            if have_old_dataset_path:
-                update_directories_to_plural(root_path, OLD_DATASETS_KEY, EntityType.DATASETS.value)
-            if have_old_model_path:
-                update_directories_to_plural(root_path, OLD_MODELS_KEY, EntityType.MODELS.value)
-            change_keys_in_config(root_path)
-        else:
-            raise Exception(output_messages['ERROR_PROJECT_NEED_BE_UPDATED'])
+    have_old_dataset_path = os.path.exists(os.path.join(root_path, V1_DATASETS_KEY)) or os.path.exists(os.path.join(
```
Comment (diego-pedroso) [inline ml_git/utils.py:349]: This variable name sounds weird. Please rename to something like `old_dataset_path_exists`. Same is valid for `have_old_model_path`

Comment (lucasncabral) [reply ml_git/utils.py]: The variable names were updated in 9398193.

**Review by diego-pedroso — APPROVED** [review]

**Pull request merged**